integrations/amazon_bedrock/tests/test_chat_generator.py

```diff
@@ -573,114 +476,6 @@ def test_pipeline_with_amazon_bedrock_chat_generator(self, model_name, tools):
-                        "model": "anthropic.claude-3-5-sonnet-20240620-v1:0",
-                        "generation_kwargs": {"temperature": 0.7},
-                        "stop_words": ["eviscerate"],
-                        "streaming_callback": "haystack.components.generators.utils.print_streaming_chunk",
-                        "boto3_config": None,
-                        "tools": [
-                            {
-                                "type": "haystack.tools.tool.Tool",
-                                "data": {
-                                    "name": "weather",
-                                    "description": "useful to determine the weather in a given location",
-                                    "parameters": {"city": {"type": "string"}},
-                                    "function": "tests.test_chat_generator.weather",
-                                },
-                            }
-                        ],
-                    },
-                }
-            },
-            "connections": [],
-        }
-
-        if not hasattr(pipeline, "_connection_type_validation"):
-            expected_dict.pop("connection_type_validation")
-
-        # add outputs_to_string, inputs_from_state and outputs_to_state tool parameters for compatibility with
-        # haystack-ai>=2.12.0
-        if hasattr(tool, "outputs_to_string"):
-            expected_dict["components"]["generator"]["init_parameters"]["tools"][0]["data"][
-                "outputs_to_string"
-            ] = tool.outputs_to_string
-        if hasattr(tool, "inputs_from_state"):
-            expected_dict["components"]["generator"]["init_parameters"]["tools"][0]["data"][
-                "inputs_from_state"
-            ] = tool.inputs_from_state
-        if hasattr(tool, "outputs_to_state"):
-            expected_dict["components"]["generator"]["init_parameters"]["tools"][0]["data"][
-                "outputs_to_state"
-            ] = tool.outputs_to_state
-
-        assert pipeline_dict == expected_dict
-
-        # Test YAML serialization/deserialization
-        pipeline_yaml = pipeline.dumps()
-        new_pipeline = Pipeline.loads(pipeline_yaml)
-        assert new_pipeline == pipeline
-
-        # Verify the loaded pipeline's generator has the same configuration
-        loaded_generator = new_pipeline.get_component("generator")
-        assert loaded_generator.model == generator.model
-        assert loaded_generator.generation_kwargs == generator.generation_kwargs
-        assert loaded_generator.streaming_callback == generator.streaming_callback
-        assert len(loaded_generator.tools) == len(generator.tools)
-        assert loaded_generator.tools[0].name == generator.tools[0].name
-        assert loaded_generator.tools[0].description == generator.tools[0].description
-        assert loaded_generator.tools[0].parameters == generator.tools[0].parameters
-
 
 class TestAmazonBedrockChatGeneratorAsyncInference:
```

Comment:
if every test case in this class is an integration test, you can apply the following decorators to the class:
- `@pytest.mark.integration`
- `@pytest.mark.skipif`